homeassistant/components/miraie_ac/climate.py

```diff
@@ -0,0 +1,303 @@
+            return SWING_BOTH
+
+        if MirAIeSwingMode.AUTO not in (v_swing, h_swing):
+            return SWING_OFF
+
+        if v_swing == MirAIeSwingMode.AUTO:
+            return SWING_VERTICAL
+
+        return SWING_HORIZONTAL
+
+    def set_temperature(self, **kwargs: Any) -> None:
+        """Set target temperature."""
+        if (temperature := kwargs.get(ATTR_TEMPERATURE)) is None:
+            raise ValueError("No Target Temperature provided")
+
+        self.device.set_temperature(temperature)
+        _LOGGER.debug(
+            "%s - Target Temperature set: %s",
+            self._friendly_name,
+            temperature,
+        )
+
+    def set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set HVAC mode."""
+        _LOGGER.debug(
+            "%s - HVAC Mode set: %s",
+            self._friendly_name,
+            hvac_mode.value,
+        )
+
+        if hvac_mode == HVACMode.OFF:
+            self.device.turn_off()
+            return
+
+        if self.device.status.power_mode == MirAIePowerMode.OFF:
+            self.device.turn_on()
+
+        self.device.set_hvac_mode(HVAC_MODE_MAP_TO_MIRAIE[hvac_mode])
+
+    def set_fan_mode(self, fan_mode: str) -> None:
+        """Set Fan mode."""
+        self.device.set_fan_mode(FAN_MODE_MAP_TO_MIRAIE[fan_mode])
+        _LOGGER.debug(
+            "%s fan mode set: %s",
+            self._friendly_name,
+            fan_mode,
+        )
+
+    def set_swing_mode(self, swing_mode: str) -> None:
+        """Set Swing mode."""
+
+        if swing_mode == SWING_BOTH:
+            self.device.set_vertical_swing_mode(MirAIeSwingMode.AUTO)
+            self.device.set_horizontal_swing_mode(MirAIeSwingMode.AUTO)
+        elif swing_mode == SWING_HORIZONTAL:
+            self.device.set_vertical_swing_mode(MirAIeSwingMode.ONE)
+            self.device.set_horizontal_swing_mode(MirAIeSwingMode.AUTO)
+        elif swing_mode == SWING_VERTICAL:
+            self.device.set_vertical_swing_mode(MirAIeSwingMode.AUTO)
+            self.device.set_horizontal_swing_mode(MirAIeSwingMode.ONE)
+        elif swing_mode == SWING_OFF:
+            self.device.set_vertical_swing_mode(MirAIeSwingMode.ONE)
+            self.device.set_horizontal_swing_mode(MirAIeSwingMode.ONE)
+
+        _LOGGER.debug(
+            "%s swing mode set: %s",
+            self._friendly_name,
+            swing_mode,
+        )
+
+    def set_preset_mode(self, preset_mode: str) -> None:
+        """Set Preset mode."""
+        self.device.set_preset_mode(PRESET_MODE_MAP_TO_MIRAIE[preset_mode])
+
+        _LOGGER.debug(
+            "%s preset mode set: %s",
+            self._friendly_name,
+            preset_mode,
+        )
+
+    def entity_state_changed_callback(self):
+        """Device status has changed, notify Hass that the entity state must be updated."""
+        _LOGGER.debug("%s - Hass State Updated", self._friendly_name)
+        self.hass.loop.call_soon_threadsafe(self.async_write_ha_state)
```

Comment:
Some of the above properties can be set directly here instead.

Reply:
Hi @edenhaus, could you elaborate? I am not sure what is the issue is. Thanks.

Reply:
Instead of creating all the properties above, you can directly set the variable.
Example temperature:
`self._attr_current_temperature = self.device.status.room_temp`

Reply:
I want to have the debug logging so as to see when new updates are pushed. I've seen delays occasionally. 

I guess I could remove the logging, but Is there an advantage to doing it this way? - just trying to understand the reason behind this recommendation

Reply:
With `async_write_ha_state`, you inform the state machine about the new state. I don't see any advantage of logging the access of the properties too. How big are these delays?
Please note, that you call `async_write_ha_state` within `call_soon_threadsafe` and it's written in the docs:

> Schedule the callback [callback](https://docs.python.org/3/glossary.html#term-callback) to be called with args arguments at the next iteration of the event loop.

Reply:
Ok, I've moved all the updates to an internal method.